src/twisted/test/test_failure.py

```diff
@@ -65,6 +65,9 @@ def test_trapRaisesWrappedException(self):
         expected types, L{failure.Failure.trap} raises the wrapped
         C{Exception}.
         """
+        if not _PY3:
+            raise SkipTest('Only relevant on Python 3.')
```

Comment:
It would be nice if this message were more informative.  Twisted raises the wrapped exception in Python 3 because before [5519](https://github.com/twisted/twisted/pull/843/files), `Failure` did not inherit from `BaseException`, and Python 3 can only `raise` instances of `BaseException`.  Now that `Failure` _is_ a `BaseException`, this behavior persists because it's the API, and as this PR demonstrates, it's bad to break that API.  Even just a `@see` directed at [the documentation](http://twisted.readthedocs.io/en/latest/core/howto/python3.html#twisted-python-failure) would be great.